File: 0101-symmetric-tree/0101-symmetric-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) return true;
        return check(root, root);
    }
    
    private boolean check(TreeNode n1, TreeNode n2){
        if(n1==null && n2==null){
            return true;
        }
        if(n1==null) return false;
        if(n2==null) return false;
        if(n1.val == n2.val){
            return check(n1.left, n2.right) && check(n1.right, n2.left);
        }else{
            return false;
        }
    }
}

File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        def check(p, q) -> bool:
            if p==None and q==None:
                return True
            
            if p==None or q==None:
                return False

            if p.val == q.val:
                return check(p.left, q.right) and check(p.right, q.left)
            else:
                return False
            
        return check(root, root)